### ISSUE TITLE
`no-missing-blank-lines` should allow for tight lists

### ISSUE THREAD

**Issue by josiahsprague** [wooorm/remark-lint#85]
I'm trying to add a table of contents which has multiple levels to a readme file. It should look like this;

```
- [Installation](#installation)
- [Usage](#usage)
  - [Options](#options)
  - [Reports](#reports)
  - [Custom Dictionary](#custom-dictionary)
  - [Changing Default Rules](#changing-default-rules)
- [Writing Standards](#writing-standards)
- [Troubleshooting](#troubleshooting)
- [Contribute](#contribute)
- [License](#license)
```

But when I run in through `remark-lint`, it says "Missing blank line before block node  no-missing-blank-lines". To fix the warning, I have to turn off the rule (not ideal, because I do want to check for proper spacing between nodes) or format the block like this;

```
- [Installation](#installation)
- [Usage](#usage)
  
  - [Options](#options)
  - [Reports](#reports)
  - [Custom Dictionary](#custom-dictionary)
  - [Changing Default Rules](#changing-default-rules)
- [Writing Standards](#writing-standards)
- [Troubleshooting](#troubleshooting)
- [Contribute](#contribute)
- [License](#license)
```

Which renders incorrectly on GitHub and npm;

- [Installation](#installation)
- [Usage](#usage)
  
  - [Options](#options)
  - [Reports](#reports)
  - [Custom Dictionary](#custom-dictionary)
  - [Changing Default Rules](#changing-default-rules)
- [Writing Standards](#writing-standards)
- [Troubleshooting](#troubleshooting)
- [Contribute](#contribute)
- [License](#license)

This issue is similar to #79

**Comment by josiahsprague** [comment]
For now I'm turning off this rule in `quality-docs`, https://github.com/SparkartGroupInc/quality-docs/blob/master/sample-rules-override.json#L11

Hopefully I can find a way to re-enable it soon.

**Comment by wooorm** [comment]
It’s not a bug, it’s exactly what this rule is for. In that list item, there’s a paragraph and a list (which are block-levels), so `no-missing-blank-lines` thinks there should be a blank between them.

> Hopefully I can find a way to re-enable it soon.

Use comments? Or PR support for [passing in exceptions](https://github.com/wooorm/remark-lint/blob/d7abf05c857977a03f1b7956a65666c46db35831/lib/rules/no-missing-blank-lines.js#L41) (I’d appreciate it!)

**Comment by bkeepers** [comment]
I just ran into this as well. Most markdown implementations have different behavior for list items depending on the white space around them. If there is a blank line between list items, then they will wrap the list item in a paragraph (as @josiahsprague's example shows above). If there is not  blank line between list items, then there will be no paragraph tag.

I understand that this complicates the implementation, but I feel like ignoring this rule for nested lists would be consistent with real-world use of markdown.

**Comment by wooorm** [comment]
@bkeepers Appreciate the input. Re-opening so we can track it.

From what I gather, this issue relates to an exception for a paragraph followed by a list, in a list item?

So:

```md
- Paragraph
  - Nested
```

Yields:

- Paragraph
  - Nested

...and is allowed.

**Comment by bkeepers** [comment]
> From what I gather, this issue relates to an exception for a paragraph followed by a list, in a list item?

Yep. Thanks for taking this into consideration. I looked into implementing it, but I'm still getting familiar with the remark code and it wasn't quite obvious to me the best way to solve it.

**Comment by wooorm** [comment]
Thanks both for convincing me this was needed. 👍 I’ll release it in a minute.

@bkeepers Appreciate you looking in. I think it’s a pretty hard project to get into, lots of moving parts. I’d like to make it easier to get into but I’m not really sure how to. If you’ve got any pointers I’d love to hear them (as issues, or [Gitter](https://gitter.im/wooorm/remark), or whatnot).

Cheers!
